src/main/scala/mesosphere/marathon/state/MarathonStore.scala

```diff
@@ -113,7 +113,9 @@ class MarathonStore[S <: MarathonState[_, S]](
       Some(newState().mergeFromProto(bytes))
     }
     catch {
-      case e: InvalidProtocolBufferException => None
+      case e: InvalidProtocolBufferException =>
```

Comment:
Ok, passing stateFromBytes the key for debugging is technically simple. From talking to you, I understood that you do not like that. I can follow you, even if I do not agree.

The alternative is to perform the error handling in the calling methods: modify and fetch.

Right now, they have really strange error handling semantics. If a key does not exist, an exception is thrown. If the data retrieved for the key cannot be read, the error is ignored and either None or a default empty object is returned.

I will change that but that of course will pretty much extend the scope of this patch.